test/e2e/features_test.go

```diff
@@ -528,12 +534,84 @@ func TestDeployAllInOneDBLESSNoLoadBalancer(t *testing.T) {
 	verifyHTTPRoute(ctx, t, env)
 }
 
+func TestDeployAllInOneDBLESSManagedGateway(t *testing.T) {
+	t.Log("configuring all-in-one-dbless.yaml manifest test")
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	t.Log("building test cluster and environment")
+	configFile, err := os.CreateTemp(os.TempDir(), "test-default-ingress-class")
+	require.NoError(t, err)
+	defer os.Remove(configFile.Name())
+	defer configFile.Close()
+	written, err := configFile.Write([]byte(webhookKINDConfig))
+	require.NoError(t, err)
+	require.Equal(t, len(webhookKINDConfig), written)
+
+	clusterBuilder := kind.NewBuilder()
+	clusterBuilder.WithConfig(configFile.Name())
+	if clusterVersionStr != "" {
+		clusterVersion, err := semver.ParseTolerant(clusterVersionStr)
+		require.NoError(t, err)
+		clusterBuilder.WithClusterVersion(clusterVersion)
+	}
+	cluster, err := clusterBuilder.Build(ctx)
+	require.NoError(t, err)
+	addons := []clusters.Addon{}
+	addons = append(addons, metallb.New())
+	if b, err := loadimage.NewBuilder().WithImage(imageLoad); err == nil {
+		addons = append(addons, b.Build())
+	}
+	builder := environments.NewBuilder().WithExistingCluster(cluster).WithAddons(addons...)
+	env, err := builder.Build(ctx)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, env.Cleanup(ctx))
+	}()
+
+	t.Log("deploying kong components")
+	manifest, err := getTestManifest(t, dblessPath)
+	require.NoError(t, err)
+	deployment := deployKong(ctx, t, env, manifest)
+
+	// ensure that Gateways with no addresses come online and start ingesting routes
+	t.Logf("deploying Gateway APIs CRDs from %s", consts.GatewayExperimentalCRDsKustomizeURL)
+	require.NoError(t, clusters.KustomizeDeployForCluster(ctx, env.Cluster(), consts.GatewayExperimentalCRDsKustomizeURL))
+
+	controllerName := "example.com/test-controller"
+	deployment, err = env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).Get(ctx, deployment.Name, metav1.GetOptions{})
+	require.NoError(t, err)
+	t.Log("updating kong deployment to enable Gateway feature gate and customize the controller-name")
+	for i, container := range deployment.Spec.Template.Spec.Containers {
+		if container.Name == "ingress-controller" {
+			deployment.Spec.Template.Spec.Containers[i].Env = append(deployment.Spec.Template.Spec.Containers[i].Env,
+				corev1.EnvVar{Name: "CONTROLLER_FEATURE_GATES", Value: "GatewayAlpha=true"})
+			deployment.Spec.Template.Spec.Containers[i].Args = append(deployment.Spec.Template.Spec.Containers[i].Args, fmt.Sprintf("--controller-name=%s", controllerName))
+		}
+	}
+	_, err = env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).Update(ctx,
+		deployment, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	t.Log("verifying controller updates associated Gateway resoures")
+	gwc := deployGatewayClass(ctx, t, env, controllerName)
+	gw := deployGateway(ctx, t, env, gwc.Name)
+	deployHTTPRoute(ctx, t, env, gw)
+	verifyHTTPRoute(ctx, t, env)
+}
```

Comment:
I'm a little confused by this test, it's not _actually_ testing a **managed** gateway right?

I just realized something else: the HTTPRoute should not be able to attach without the `GatewayClass` status becoming ready for use... do we have contention between the KGO and the KIC where _both_ are trying to manage the status of the `GatewayClass`? We may need to add the `unmanaged` annotation to the `GatewayClass` as a means to not have both controllers colliding on those.

Reply:
As far as I can see in KGO we never update the GatewayClass status, there shouldn't be any collision

Reply:
I'm not convinced too about this test. I want to test managed gateways (I changed the helper), but it is just an attempt; it is not ready yet. I'm afraid that without an actor reconciling the Gateway status, the HTTPRoute could never get ready. So probably, we should perform an e2e test in KGO to test this functionality for real.

Reply:
I dropped the test

Reply:
I'll consider this comment resolved, some relevant follow up added in https://github.com/Kong/kubernetes-ingress-controller/pull/2917#pullrequestreview-1102604851